BrightID/src/components/Recovery/RecoveringConnectionCard.js

```diff
@@ -25,24 +30,24 @@ class RecoveryConnectionCard extends React.PureComponent<Props> {
     };
   }
   handleConnectionSelect = async () => {
-    const { t } = this.props;
+    const { t, id, aesKey, navigation } = this.props;
     try {
-      const { signingKey, timestamp } = parseRecoveryQr(
-        this.props.recoveryRequestCode,
-      );
-
-      await backupApi.setSig({ id: this.props.id, timestamp, signingKey });
-
-      Alert.alert(
-        t('common.alert.info'),
-        t('restore.alert.text.requestRecovering'),
-        [
-          {
-            text: t('common.alert.ok'),
-            onPress: () => this.props.navigation.navigate('Home'),
-          },
-        ],
-      );
+      const ipAddress = await api.ip();
```

Comment:
There is a lot of network activity going on in this handler without visible user feedback. On bad network condition or when having much data to transfer the user will not know if the app has crashed or is still working.

I think we need:
- Button should be disabled after pressed, so user can not trigger multiple parallel executions
- Screen should show a waitcursor/spinner, so it is clear for the user that something is going on (Best case would be a nice progress bar, possibly reusing `Tasks/TasksProgress.js`?)

Reply:
We already have a spinner added there that shows waiting to users while their app is uploading the mutual info.
Disabling the button is a good idea too.